src/broker/index.js

```diff
@@ -39,21 +41,31 @@ module.exports = class Broker {
    * @returns {Promise}
    */
   async connect() {
-    this.authenticated = false
-    await this.connection.connect()
+    try {
+      await this.lock.acquire()
 
-    if (!this.versions) {
-      this.versions = await this.apiVersions()
-    }
+      if (this.isConnected()) {
+        return false
```

Comment:
I don't think I agree with this. I don't think anyone outside of the broker should be aware of whether you connected or if you became connected because some other request caused you to become connected.